pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -108,6 +108,9 @@ public class PinotQueryResource {
   public String handlePostSql(String requestJsonStr, @Context HttpHeaders httpHeaders) {
     try {
       JsonNode requestJson = JsonUtils.stringToJsonNode(requestJsonStr);
+      if (!requestJson.has("sql")) {
+        throw new IllegalStateException("JSON Payload is missing the query string field 'sql'");
```

Comment:
should not throw IllegalStateException --> this will make the return code 500 INTERNAL_ERROR

instead we should throw a ProcessingException with a client-request error code 
e.g.
```
QueryException.getException(JSON_PARSING_ERROR_CODE, "...")
```

Reply:
Done